libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -686,16 +688,35 @@ class MapboxNavigation(
         tripSession.updateSensorEvent(sensorEvent)
     }
 
-    /**
-     * Updates the configuration to enable or disable the extended kalman filter (EKF).
-     *
-     * @param useEKF the new value for EKF
-     */
-    fun useExtendedKalmanFilter(useEKF: Boolean) {
-        tripSession.useExtendedKalmanFilter(useEKF)
+    private fun configureRouter() {
+        with(navigationOptions) {
+            ThreadController.getMainScopeAndRootJob().scope.launch {
+                val offlineFilesPath = OnboardRouterFiles(applicationContext, logger)
+                    .absolutePath(onboardRouterOptions)
+                offlineFilesPath?.let { path ->
+                    val routerParams = RouterParams(
+                        path,
+                        null,
+                        null,
+                        THREADS_COUNT,
+                        TileEndpointConfiguration(
+                            onboardRouterOptions.tilesUri.toString(),
+                            onboardRouterOptions.tilesVersion,
+                            accessToken ?: "",
+                            USER_AGENT,
+                            "",
+                            NativeSkuTokenProvider(MapboxNavigationAccounts.getInstance(applicationContext))
+                        )
+                    )
+                    navigator.configureRouter(routerParams)
```

Comment:
Capturing from the NN team, `configureRouter` is soon going to be moved to the constructor as well, so it's good to see it moved out of the onboard router and have it ready for that change :+1:

Reply:
This answers my question, thanks!

Considering that MapboxNavigation is a god class and is difficult to add tests specific to any feature, we should consider other ways to delegate this logic.